tools/README.md

```diff
@@ -35,19 +40,70 @@ $ python tools/tokenizer.py --text_input_path raw_data.txt --bin_output_path cn/
 其中，`cn`表示中文数据集；`en`表示英文数据集；`code`表示代码数据集；`ja`表示日语数据集；`ar`表示阿拉伯语数据集；`kaoshi`表示考试数据集。
 
 生成的bin文件的格式如下：
+
 ```python
 {"tokens": [73075, 75302, 69522, 69022, 98899, 67713, 68015, 81269, 74637, 75445, 99157]}
 {"tokens": [69469, 60355, 73026, 68524, 60846, 61844, 98899, 67775, 79241, 98899, 67713, 67800, 67453, 67838, 99157]}
 {"tokens": [68057, 79017, 60378, 68014, 98899, 67713, 67990, 68015, 70381, 67428, 61003, 67622, 99157]}
 ```
+
 `bin`文件中的每一行均对应原始数据集中的每一个句子，表示每个句子的`token`（下文将用sequence指定）。
 
 生成的`meta`文件的格式如下：
+
 ```bash
 (0, 11), (90, 15), (208, 13)
 ```
+
 在`meta`文件中，每个元组对应着`bin`文件中每一个`sequence`的元信息。其中，元组的第一个元素表示每个`sequence`在所有`sequence`中的`starting index`，第二个元素表示每个`sequence`中有多少个`tokens`。
 
 例如，对于第一个`sequence`，`starting index`为 0，有 11 个`tokens`；对于第二个`sequence`，由于第一个`sequence`转换为`string`后的长度为`89`，因此它的`starting index`为 90，有 15 个`tokens`。
 
-`json`和`jsonl`类型的文件的`bin`和`meta`文件格式和`txt`一致，此处不再赘叙。
\ No newline at end of file
+`json`和`jsonl`类型的文件的`bin`和`meta`文件格式和`txt`一致，此处不再赘叙。
+
+# pal_inference.py
+
+在 [GSM8K](https://huggingface.co/datasets/gsm8k) 数据集上使用 [PAL](https://github.com/reasoning-machines/pal) 范式推理，使模型编写代码并通过 Python 解释器执行来解决数学问题。其用法如下：
+
+```python
+# 用法:
+python pal_inference.py <model> <out_dir> [--dataset <dataset>] [--max_length <length>] [--top_p <threshold>] [--temperature <temp>] [--time_out <time>] [--verbose, -v] [--append, -a]
+
+# 参数:
+# <model>             用于推理的模型的路径。
+# <out_dir>           生成代码将保存在指定的输出文件夹中。
+
+# 可选参数:
+# --dataset <dataset>   用于代码生成的数据集名称（默认：gsm8k）。
+# --max_length <length> 模型最大输入toekn长度（默认：2048）。
+# --top_p <threshold>   候选token相加的概率阈值（默认：0.8）。
+# --temperature <temp>  生成过程中的采样温度（默认：1）。
```

Comment:
1 or 1.0? we may give the range of the temperature, moreover, maybe it can be simple as `-t`